Faith_15-7_BeatThat.js

```diff
@@ -0,0 +1,220 @@
+  for (var i = 1; i < diceCt; i += 1) {
+    num = num * 10 + diceArray[i];
+  }
+  return num;
+}
+//determine smallest number rolled among players by sorting array in descending order
+function smallestChosenNum(diceArray, diceCt) {
+  diceArray.sort(function (a, b) {
+    return a - b;
+  });
+  var num = diceArray[0];
+  for (var i = 1; i < diceCt; i += 1) {
+    num = num * 10 + diceArray[i];
+  }
+  return num;
+}
+var rollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var diceNum = randomInteger + 1;
+  return diceNum;
+};
+//generate n number of rolls from dice, depending how many dices you chose
+var rollDiceNum = function () {
+  for (n = 0; n < diceNum; n += 1) {
+    diceRoll = rollDice();
+    diceArray.push(diceRoll);
+  }
+};
+var resetGame = function () {
+  userNameArray = [];
+  chosenNumArray = [];
+  diceArray1 = [];
+  diceArray2 = [];
+  userName = null;
+  diceNum = null;
+  gameMode = null;
+};
+var main = function (input) {
+  //check gamemode
+  if (!gameMode) {
+    if (!input) {
+      return chooseModeMsg();
+    }
+    modeValid = checkMode(input);
+    if (!modeValid) {
+      return chooseModeMsg();
+    }
+    gameMode = input;
+    return gameModeInstructions(gameMode);
+  }
+  if (gameMode == "highest") {
+    //run game for 2 players
+    if (chosenNumArray.length >= 0 && chosenNumArray.length < 2) {
+      if (!userName) {
+        if (!input || input != 1) {
+          return `Please enter '1' to begin.`;
+        }
+        userName = input;
+        userNameArray.push(userName);
+        if (userNameArray.length == 1) {
+          return `Welcome Player ${userNameArray.length}. <br/> Enter the number of dice to roll.`;
+        } else {
+          return `Welcome Player 2. You will roll ${diceNum} dice. No input needed, click 'Submit' to continue.`;
+        }
+      }
+      if (!diceNum) {
+        if (!input || input < 2) {
+          return `Invalid input. Enter a number of dice >=2.`;
+        }
+        diceNum = input;
+        if (userNameArray.length == 1) {
+          userName = null;
+          for (n = 0; n < diceNum; n += 1) {
+            diceRoll = rollDice();
+            diceArray1.push(diceRoll);
+          }
+          chosenNum = largestChosenNum(diceArray1, diceNum);
+          chosenNumArray.push(chosenNum);
+          return `Player ${userNameArray.length}, your number is ${chosenNum}.<br/> It is now Player 2's turn.<br/> Player 2, please key in '1' to play. `;
+        } else {
+          userName = null;
+          for (n = 0; n < diceNum; n += 1) {
+            diceRoll = rollDice();
+            diceArray2.push(diceRoll);
+          }
+          chosenNum = largestChosenNum(diceArray2, diceNum);
+          chosenNumArray.push(chosenNum);
+          diceNum = null;
+          return `Player ${userNameArray.length}, your number is ${chosenNum}.<br/> Click Submit to determine the winner!`;
+        }
```

Comment:
I don't think this else block will ever run because if `diceNum` has been defined, it means that we are not at player 1 anymore - can remove if unnecessary!